Add clang-format check along with checkpatch.

psathyan opened the pull request:
The checkpatch job is updated to include clang-format check. We would be posting another message to the review with the status of the check.

The clang-format is run using `git clang-format HEAD~1`. On the node, we would be instaling `git-clang-format` to enable the check.

psathyan committed:
Add clang-format check along with checkpatch.

The checkpatch job is updated to include clang-format check. We
would be posting another message to the review with the status
of the check.

Signed-off-by: Pragadeeswaran Sathyanarayanan <pragadeeswaran.sathyanarayanan@ibm.com>

ffilz commented on jobs/scripts/checkpatch.sh:
```diff
@@ -2,33 +2,93 @@ set -o pipefail
-  ( cd $WORKSPACE/ci-tests/build_scripts/checkpatch && ./checkpatch.pl -q - || true ) | \
-  python $WORKSPACE/ci-tests/build_scripts/checkpatch/checkpatch-to-gerrit-json.py    | \
-  publish_checkpatch
+# Allow checkpatch to complete before calling clang-format
```
One comment here - we should run clang-format first, that is now our primary style check. Assuming the patch has been clang-formatted properly, MOST of the checkpatch checks will pass, there are just a few things checkpatch expects that clang-format doesn't care about.

psathyan replied:
Thank you @ffilz 

I have incorporated the suggestion with the latest commit.

ffilz commented:
One additional comment, please update:

/home/ffilz/ganesha/ci-tests/build_scripts/checkpatch/.checkpatch.conf

From the version in Ganesha

psathyan committed:
Align checkpatch conf

Ensuring the checkpatch conf is aligned to the configuration
file found in nfs-ganesha/src/scripts. This allows us to have
a consistent experience.

Signed-off-by: Pragadeeswaran Sathyanarayanan <pragadeeswaran.sathyanarayanan@ibm.com>

psathyan commented:
> One additional comment, please update:
> 
> /home/ffilz/ganesha/ci-tests/build_scripts/checkpatch/.checkpatch.conf
> 
> From the version in Ganesha

Done. The `.checkpatch.conf` is now aligned with the version in `nfs-ganesha`.
